test/src/test/java/jenkins/tasks/SimpleBuildWrapperTest.java

```diff
@@ -84,6 +86,20 @@
         }
     }
 
+    @Issue("JENKINS-27188")
+    @Test public void envPropertiesImmutable() throws Exception {
+        SpecialEnvSlave slave = new SpecialEnvSlave(tmp.getRoot(), r.createComputerLauncher(null));
+        r.jenkins.addNode(slave);
+
+        String propertyKey = "JENKINS-27188";
+        EnvVars envVars = slave.getComputer().getEnvironment();
+        envVars.put(propertyKey, "huuhaa");
+        assertTrue(envVars.containsKey(propertyKey));
+        assertFalse(slave.getComputer().getEnvironment().containsKey(propertyKey));
+
+        assertNotSame(slave.getComputer().getEnvironment(), slave.getComputer().getEnvironment());
+    }
```

Comment:
This test does not belong here. Could create a `ComputerTest` for it since I do not see any other obvious place.